scipy/signal/tests/test_signaltools.py

```diff
@@ -1093,20 +1093,13 @@ def test_types(self, dtype):
         assert_equal(signal.medfilt(in_typed).dtype, dtype)
         assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
 
-    def test_types_deprecated(self):
-        dtype = np.longdouble
-        in_typed = np.array(self.IN, dtype=dtype)
-        msg = "Using medfilt with arrays of dtype"
-        with pytest.deprecated_call(match=msg):
-            assert_equal(signal.medfilt(in_typed).dtype, dtype)
-        with pytest.deprecated_call(match=msg):
-            assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
-
-
     @pytest.mark.parametrize('dtype', [np.bool_, np.complex64, np.complex128,
-                                       np.clongdouble, np.float16,])
+                                       np.clongdouble, np.float16, np.longdouble,
+                                       np.object_])
     def test_invalid_dtypes(self, dtype):
         in_typed = np.array(self.IN, dtype=dtype)
+        if in_typed.dtype is np.float64:
+            pytest.mark.skip("Platform does not support `float128`")
```

Comment:
The combination of if-condition and skip-message makes no sense. It would be much better not to parametrize on an ambiguous type (`np.longdouble`) in the first place, but if that's unavoidable, one way to clarify would be (feel free to rewrite):
```suggestion
        if in_typed.dtype is np.float64:
            pytest.mark.skip("Platform treats `np.longdouble` as `np.float64`; "
                             "we support the latter, but not the former (in general).")
```

Reply:
Agh, the tests are still failing on windows and macos arm, neither of which I have access to so this clearly is not quite right. @ev-br would you be able to point me in the right direction here as I am getting quite confused!

Reply:
I don't think `if dtype is np.float64` is correct really. I suspect you're hitting https://github.com/scipy/scipy/pull/18341#issuecomment-1532754157, so can dispatch on `in_typed.dtype.char == 'g'` instead?

I can check it in a couple of days unless you beat me to it.

Reply:
Or just don't parametrize the test with `np.longdouble`, but rather `float96` & `float128`.

Reply:
Maybe I am misunderstanding but I don't think we can use `float128` directly here as it doesn't exist on some platforms.

Reply:
Yes, you'll need to skip it on platforms that don't have it. I made another suggestion that's IMO easier to follow, because there's no need to invert any logic (of the kind "we support everything but these types" or "we support `np.longdouble` except when it's actually `np.float64`).